piccolo_admin/endpoints.py

```diff
@@ -403,6 +412,24 @@ def __init__(
         self.description = description
         self.form_group = form_group
         self.slug = self.name.replace(" ", "-").lower()
+        for (
+            field_name,
+            field_value,
+        ) in self.pydantic_model.model_fields.items():
+            if isinstance(field_value.annotation, ENUMTYPE):
+                # update model_fields, field annotation and
+                # rebuild the model for the changes to take effect
+                pydantic_model.model_fields[field_name] = Field(
+                    json_schema_extra={
+                        "extra": {
+                            "choices": convert_enum_to_choices(
+                                field_value.annotation
+                            )
+                        }
+                    },
+                )
+                pydantic_model.model_fields[field_name].annotation = str
```

Comment:
Can't we just leave the annotation as what it was originally?

Reply:
We need to reassign the annotation type because if we don't, that annotation is `None` and that causes a form validation error. I'm using the string type because if we reassign annotation to the `Enum`, frontend don't work (working with strings because choices `display_value` and `value` are strings, and choices does not work with `Enum`). In the whole PR I tried to use Piccolo choices so that there would be as few changes as possible on the frontend. If you don't like the whole concept, feel free to delete this PR.

Reply:
Ah, yes - I forgot that the front end would get confused by the open api schema is we leave the enum in there.